cranelift/codegen/src/isa/x64/inst/emit.rs

```diff
@@ -1737,14 +1715,21 @@ pub(crate) fn emit(
             // TODO: this needs to be accounted for by the stack check, before `GrowFrame` is
             // emitted, so that the increment here is expected.
             //
-            // Decrement SP by `amount`
+            // Decrement SP and FP by `amount`
             Inst::alu_rmi_r(
                 OperandSize::Size64,
                 AluRmiROpcode::Sub,
                 RegMemImm::imm(*amount),
                 Writable::from_reg(regs::rsp()),
             )
             .emit(&[], sink, info, state);
+            Inst::alu_rmi_r(
+                OperandSize::Size64,
+                AluRmiROpcode::Sub,
+                RegMemImm::imm(*amount),
+                Writable::from_reg(regs::rbp()),
+            )
+            .emit(&[], sink, info, state);
```

Comment:
Can we debug assert that frame pointers are enabled here? Or guard this on frame pointers being enabled?

Reply:
We haven't removed the assert in `emit_return_call_common_sequence` that frame pointers are enabled, but yeah, we should add that in both `ShrinkFrame` and `GrowFrame` as well.